py_experimenter/experimenter.py

```diff
@@ -29,58 +29,101 @@ def __init__(self,
+        elif self.config['PY_EXPERIMENTER']['provider'] == 'mysql':
+            self.dbconnector = DatabaseConnectorMYSQL(self.config, credential_path)
         else:
             raise ValueError('The provider indicated in the config file is not supported')
 
         logging.info('Initialized and connected to database')
 
     def set_config_value(self, section_name: str, key: str, value: str) -> None:
         """
-        Sets a value in the configuration file. If the section does not exist, it is created.
-        :param section_name: Name of the section in the configuration file.
-        :param key: Name of the key in the configuration file.
-        :param value: Value to set.
+        Modifies the experiment configuration so that within the given `section_name` the value of the given `key` is overridden, 
+        or created if it is not existing beforehand.
+
+        :param section_name: The name of the section of the experiment configuration in which a value should be set.
+        :type section_name: str
+        :param key: The name of the key within the given section of the experiment configuration in which a value should be set.
+        :type key: str
+        :param value: The value of the key within the given section of the experiment configuration that should be set.
+        :type value: str
         """
-        if not self._config.has_section(section_name):
-            self._config.add_section(section_name)
-        self._config.set(section_name, key, value)
-        PyExperimenter._valid_configuration(self._config, self._credential_path)
+        if not self.config.has_section(section_name):
+            self.config.add_section(section_name)
+        self.config.set(section_name, key, value)
+        PyExperimenter._is_valid_configuration(self.config, self.credential_path)
 
     def get_config_value(self, section_name: str, key: str) -> str:
-        return self._config.get(section_name, key)
+        """
+        Returns the value of the experiment configuration for the given `key`. If the `key` is not contained within the section, 
+        an exception is raised.
+
+        :param section_name: The name of the section of the experiment configuration of which a value should be returned.
+        :type section_name: str
+        :param key: The name of the key within the given section of the experiment configuration of which a value should be returned
+        :type key: str
+        :return: The value of the `key` field of the experiment configuration within the section having `section_name`.
+        :rtype: str
+        :raises NoOptionError: If the `key` is not contained within the section.
+        """
+        return self.config.get(section_name, key)
 
     def has_option(self, section_name: str, key: str) -> bool:
-        return self._config.has_option(section_name, key)
+        """
+        Checks whether the experiment configuration contains the given `key` within the section called `section_name`.
```

Comment:
Checks whether the experiment configuration contains a property identified by the given 'key' within the section called 'section_name'.